src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -599,6 +652,7 @@ private void handleExit(int exitCode, OutputSupplier output) throws IOException,
                     getContext().onFailure(new AbortException("script returned exit code " + exitCode));
                 }
             }
+            listener().getLogger().close();
```

Comment:
So this call will always be on `NewlineSafeTaskListener`, right? Is this why we don't want to call `super.close()` in `NewlineSafeTaskListener.$NewlineSafeTaskListener.close()`? Ditto for the call below in `exited`. Also, if we are calling `close` here now, why didn't we call `flush` here earlier? Should we have been doing that?

Reply:
> So this call will always be on `NewlineSafeTaskListener`, right?

Right.

> Is this why we don't want to call `super.close()` in `NewlineSafeTaskListener.$NewlineSafeTaskListener.close()`?

Yes.

> if we are calling `close` here now, why didn't we call `flush` here earlier?

There is no particular need to flush until the end of the step (and `close` flushes).

Reply:
> There is no particular need to flush until the end of the step (and close flushes).

My question was unclear, but my confusion is about why we weren't calling `flush` before this PR, since it looks like this is the end of the step.

Reply:
Oh, I see! We _were_ calling `flush` before, in watch mode, in `exited`. In non-watch mode, it did not really matter since the log messages were coming from the master, and `DefaultStepContext.getListener` always `close`s the per-step listener when the step finishes.